internal/repositories/rewards.go

```diff
@@ -0,0 +1,174 @@
+package repositories
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	gmodel "github.com/DIMO-Network/identity-api/graph/model"
+	"github.com/DIMO-Network/identity-api/internal/helpers"
+	"github.com/DIMO-Network/identity-api/models"
+	"github.com/DIMO-Network/shared/dbtypes"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+	"github.com/volatiletech/sqlboiler/v4/types"
+)
+
+type RewardsCursor struct {
+	Week      int
+	VehicleID int
+}
+
+func RewardToAPI(reward models.Reward) gmodel.Earning {
+	stEarn := dbtypes.NullDecimalToInt(reward.StreakEarnings)
+	adEarn := dbtypes.NullDecimalToInt(reward.AftermarketEarnings)
+	syEarn := dbtypes.NullDecimalToInt(reward.SyntheticEarnings)
+
+	return gmodel.Earning{
+		Week:                    reward.IssuanceWeek,
+		Beneficiary:             common.BytesToAddress(reward.ReceivedByAddress.Bytes),
+		ConnectionStreak:        reward.ConnectionStreak.Ptr(),
+		StreakTokens:            stEarn,
+		AftermarketDeviceID:     reward.AftermarketTokenID.Ptr(),
+		AftermarketDeviceTokens: adEarn,
+		SyntheticDeviceID:       reward.SyntheticTokenID.Ptr(),
+		SyntheticDeviceTokens:   syEarn,
+		SentAt:                  reward.EarnedAt,
+		VehicleID:               reward.VehicleID,
+	}
+}
+
+func (r *Repository) GetEarningsByVehicleID(ctx context.Context, tokenID int) (*gmodel.VehicleEarnings, error) {
+	type rewardStats struct {
+		TokenSum   types.NullDecimal `boil:"token_sum"`
+		TotalCount int               `boil:"total_count"`
+	}
+	var stats rewardStats
+
+	limit := new(int)
+	*limit = 100
+
+	err := models.Rewards(
+		qm.Select(
+			fmt.Sprintf(
+				`sum(%s + %s + %s) as token_sum`,
+				models.RewardColumns.StreakEarnings, models.RewardColumns.AftermarketEarnings, models.RewardColumns.SyntheticEarnings,
+			),
+			"count(*) as total_count",
+		),
+		models.RewardWhere.VehicleID.EQ(tokenID),
+	).Bind(ctx, r.pdb.DBS().Reader, &stats)
+	if err != nil {
+		return nil, err
+	}
+
+	if stats.TokenSum.IsZero() {
+		return &gmodel.VehicleEarnings{
+			TotalTokens: &big.Int{},
+			History:     &gmodel.EarningsConnection{},
+			VehicleID:   tokenID,
+		}, nil
+	}
+
+	earningsConn := &gmodel.EarningsConnection{
+		TotalCount: stats.TotalCount,
+	}
+
+	return &gmodel.VehicleEarnings{
+		TotalTokens: dbtypes.NullDecimalToInt(stats.TokenSum),
+		History:     earningsConn,
+		VehicleID:   tokenID,
+	}, nil
+}
+
+func (r *Repository) PaginateVehicleEarningsByID(ctx context.Context, vehicleEarnings *gmodel.VehicleEarnings, first *int, after *string, last *int, before *string) (*gmodel.EarningsConnection, error) {
+	limit, err := helpers.ValidateFirstLast(first, last, maxPageSize)
+	if err != nil {
+		return nil, err
+	}
+
+	queryMods := []qm.QueryMod{
+		models.RewardWhere.VehicleID.EQ(vehicleEarnings.VehicleID),
+	}
+
+	if after != nil {
+		afterID, err := helpers.CursorToID(*after)
```

Comment:
It looks like the cursors are the weeks now. This is fine for by-vehicle and by-device, but we will probably have to do something different when we look at by-user.